### PR TITLE
Added new class implementing DataSource to read CSV files

### PR DESCRIPTION
**Pull request by albertsun**
Pretty self explanatory hopefully. Can let people get up and running making their own graphs faster by letting them read in their own data in a really simple way.

### CODE REVIEW

File: CSVDataSource.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  public Layer[] make(int a, int b) {
+    int numLayers = data.length;
+    Layer[] layers = new Layer[numLayers];
+
+    for (int i = 0; i < numLayers; i++) {
+      String[] fields  = data[i].split(",");
+      int sizeArrayLength  = fields.length - 1;
+      String name      = fields[0];
+      float[] size     = new float[sizeArrayLength];
```
Comment (codyaray) [inline CSVDataSource.java:31]: This array is going unused, as immediately below you're calling a function which is returning a _new_ array and assigning to this variable. Collapse these calls into one:

```
    float[] size = makeDataArray(fields);
```

and then delete the `sizeArrayLength` declaration above.

Comment (albertsun) [inline CSVDataSource.java:31]: Aye, good point. I don't write much Java code. This place seems abandoned though.

Comment (codyaray) [inline CSVDataSource.java:31]: Yeah, I think this was mainly a research project that's gone dormant. Not sure its much used anymore though. If Lee Byron's not accepting new changes, best you can do is host a fork with your changes so others can still use them. Maybe send him an email? Not sure if GH does that for these edits. I only see them in the notifications section when I log in (frequently).

**Comment by leebyron** [comment]
Merged in e7370a65f95dc50eb190c1cef7ae4d89e42faa5c

**Pull request closed without merging**